oresat_configs/__init__.py

```diff
@@ -40,47 +38,42 @@ class Card:
     """Optional child node name. Useful for CFC cards."""
 
 
-class OreSatConfig:
-    """All the configs for an OreSat mission."""
+def cards_from_csv(oresat: Consts) -> dict[str, Card]:
+    """Turns cards.csv into a dict of names->Cards, filtered by the current mission"""
 
-    CARD_CONFIG_PATHS = {
-        OreSatId.ORESAT0: ORESAT0_CARD_CONFIGS_PATH,
-        OreSatId.ORESAT0_5: ORESAT0_5_CARD_CONFIGS_PATH,
-        OreSatId.ORESAT1: ORESAT1_CARD_CONFIGS_PATH,
-    }
+    file_path = f"{os.path.dirname(os.path.abspath(__file__))}/cards.csv"
+    with open(file_path, "r") as f:
```

Comment:
Please add either safeguards in code or corresponding docs as to what the structure of the CSV is expected to be.

Reply:
It'll `KeyError` when it doesn't find the necessary columns. Are you asking for a more descriptive error? I'm unsure what you're intending by safeguard here.

The structure of the CSV is expected to be whatever is in `cards.csv`. As far as I can tell there should only ever be one of those. If a column is removed it'll `KeyError`, if a column is added, you won't find it in the related `Card` until you explicitly add it to `Card`, and `Card` won't let you construct itself until it gets that new column.

Reply:
Right, that's what worries me- some time down the line we update `cards.csv` and it'll start throwing key errors thus crashing any generators.

Simply allowing KeyErrors to pass is not graceful IMO and likely to be a source of frustration down the road. We should either throw a useful error like *"Cards.csv is missing \_ expected columns"** or *"Has \_ unexpected column"* or something.